src/System.Collections.Specialized/src/System/Collections/Specialized/OrderedDictionary.cs

```diff
@@ -81,7 +81,11 @@ public int Count
         {
             get
             {
-                return objectsArray.Count;
+                if (_objectsArray == null)
```

Comment:
The underscore distinction between the field and the property is quite subtle. Would it make sense to rename `objectsTable` and `objectsArray` to eg `GetOrCreateObjectsTable()` and `GetOrCreateObjectsArray()` perhaps?

Reply:
Another pattern, perhaps easier to read/review, could be to add calls to `void EnsureObjectsTable()` and `void EnsureObjectsArray()` that return void and therefore all access could consistently be through the fields.

Reply:
I prefer the latter. Thoughts?

Reply:
EnsureObjectsTable() and void EnsureObjectsArray() was implemented.